### ISSUE TITLE
what does the normalization operation mean in 136&137 line of acfnet.py?

### ISSUE THREAD

**Issue by jeou** [zrl4836/ACFNet#4]
As mentioned in title. I didn't see this operation in paper. Could you show me some guidance?
https://github.com/zrl4836/ACFNet/blob/ceba02e93fbc32fbad123c5d052b87ccf3ce4f56/networks/acfnet.py#L134-L137

**Comment by zrl4836** [comment]
Prevent loss divergence during training

**Comment by qiulesun** [comment]
@zrl4836 Before conducting softmax operation on x, we usually subtract the maximum value of x. I see that [energy_new = torch.max(energy, -1, keepdim=True)[0].expand_as(energy)-energy](https://github.com/zrl4836/ACFNet/blob/master/networks/acfnet.py#L136)  rather than energy_new = energy- torch.max(energy, -1, keepdim=True)[0].expand_as(energy) followed by softmax op. Therefore, the weights of each channel are reverse based on the two different op before softmax.
